treon/treon.py

```diff
@@ -107,24 +107,40 @@ def print_test_collection(notebooks):
 
 
 def get_notebooks_to_test(args):
-    path = args['PATH'] or os.getcwd()
-    result = []
-
-    if os.path.isdir(path):
-        LOG.info('Recursively scanning %s for notebooks...', path)
-        path = os.path.join(path, '')  # adds trailing slash (/) if it's missing
-        glob_path = path + '**/*.ipynb'
-        result = glob.glob(glob_path, recursive=True)
-    elif os.path.isfile(path):
-        if path.lower().endswith('.ipynb'):
-            LOG.debug('Testing notebook %s', path)
-            result = [path]
-        else:
-            sys.exit('{path} is not a Notebook'.format(path=path))
+    path = pathlib.Path(args['PATH'] or os.getcwd())
+
+    if path.is_file():
+        if path.suffix == '.ipynb':
+            return [path.as_posix()]
+        sys.exit("{path} is not a notebook".format(path=path))
+    elif path.is_dir():
+        LOG.info("Recursively scanning %s for notebooks...", path)
+        ignored = build_ignore_list(path)
+        notebooks = filter(lambda nb: nb not in ignored, path.glob('**/*.ipynb'))
```

Comment:
Ignore paths are not working recursively. 

E.g. Even if I have following rule in .treonignore 
`site/en/r2/*`

all these following files are not-ignored 
```
/workspace/treon/tmp/docs/site/en/r2/tutorials/eager/custom_layers.ipynb
/workspace/treon/tmp/docs/site/en/r2/tutorials/load_data/images.ipynb
...
```

Reply:
Ah, good catch. It looks like `pathlib` isn't working [predictably](https://bugs.python.org/issue26115)...

Reply:
This is still not working for me. 

I have following rules in .treonignore -

```
site/en/r2/*
site/en/r2/tutorials/load_data/tf_records.ipynb
```

and the following files are still not ignored

```
'/workspace/treon/tmp/docs/site/en/r2/tutorials/eager/custom_layers.ipynb', '/workspace/treon/tmp/docs/site/en/r2/tutorials/load_data/images.ipynb', '/workspace/treon/tmp/docs/site/en/r2/tutorials/load_data/tf_records.ipynb'
```